Security Policy feature enhancement.

chshu opened the pull request:
* Initialize the Security Policy Tlv in Active Dataset

* Apply Security Policy configuration from Active Dataset

* Do not include NetworkMasterKey Tlv in MGMT_GET.rsp if Security Policy 'O' bit is disabled

* THCI: add startNativeCommissioner() and sendBeacons() support

It helps to pass Certification test Leader_5_8_4:
[Leader_5_8_4.zip](https://github.com/openthread/openthread/files/577798/Leader_5_8_4.zip)

jwhui commented on src/core/thread/key_manager.hpp:
```diff
@@ -251,16 +264,27 @@ class KeyManager
+    /**
+     * This method returns the Security Policy Flags.
+     *
+     * The Security Policy Flags specifies network administrator preferences for which security-related operations are allowed or disallowed
```
nit: limit line width to 120 characters

jwhui commented on src/core/thread/meshcop_dataset_manager.cpp:
```diff
@@ -831,6 +856,11 @@ void ActiveDataset::StartLeader(void)
         dataset.mIsPanIdSet = true;
 
+        // Security Policy
+        dataset.mSecurityPolicy.mRotationTime = KeyManager::kDefaultKeyRotationTime;
```
Replace `KeyManager::kDefaultKeyRotationTime` with `mKeyRotationTime`?

jwhui commented on src/core/thread/meshcop_dataset_manager.cpp:
```diff
@@ -831,6 +856,11 @@ void ActiveDataset::StartLeader(void)
 
+        // Security Policy
+        dataset.mSecurityPolicy.mRotationTime = KeyManager::kDefaultKeyRotationTime;
+        dataset.mSecurityPolicy.mFlags = 0xff;
```
Replace `0xff` with `mSecurityPolicyFlags`?

chshu commented:
@jwhui , I have addressed your comments with the updated commit, thanks.

And I just found a parameter mismatch in spec: `thrKeyRotation` is defined with type uint32 in section 7.1.2, but in section 8.10.1.15 the space reserved for it is 16 bits. I will file the JIRA to Thread Group.
